tutorial/Flyer_Basics.ipynb

```diff
@@ -0,0 +1,985 @@
+    "\n",
+    "* Create a logger instance in case we want to investigate internal details as our code runs.\n",
+    "* Create an instance of the BlueSky RunEngine.\n",
+    "* Create an instance of the databroker using the `mongodb_config.yml` file on the local machine\n",
+    "* Arrange for the databroker to receive all events from the RunEngine\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "0"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import logging\n",
+    "import threading\n",
+    "import time\n",
+    "\n",
+    "import ophyd\n",
+    "import bluesky\n",
+    "import bluesky.plans\n",
+    "import databroker\n",
+    "\n",
+    "logger = logging.getLogger()\n",
+    "RE = bluesky.RunEngine({})\n",
+    "db = databroker.Broker.from_config(databroker.temp_config())\n",
+    "RE.subscribe(db.insert)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "### Bare Minimum Requirements for a Flyer <a class=\"anchor\" id=\"flyer-requirements\" />\n",
+    "\n",
+    "In BlueSky, a [Flyer](http://nsls-ii.github.io/bluesky/async.html?highlight=flyer#flying) is an `ophyd.Device` that meets the Flyer interface, which has three methods:\n",
+    "\n",
+    "1. Kickoff - begin accumulating data\n",
+    "1. Complete - BlueSky tells the Flyer that BlueSky is ready to receive data\n",
+    "1. Collect - the device provides the data to BlueSky\n",
+    "\n",
+    "The first two methods [must return](http://nsls-ii.github.io/bluesky/hardware.html?highlight=flyer#kickoff) an instance of `ophyd.DeviceStatus` (a.k.a. a *status* object).  \n",
+    "\n",
+    "The `collect()` method requires a companion `describe_collect()` that informs the RunEngine what kind of data to expect from `collect()`.\n",
+    "\n",
+    "This example (which does absolutely nothing) meets the bare minimum requirement."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "DeviceStatus(device=flyer, done=True, success=True)\n",
+      "[{'data': {}, 'timestamps': {}, 'time': 1524854056.340829}]\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "('ab9f6cd6-2ed2-420c-8dbb-d65adedb0add',)"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "class BareMinimumFlyer(ophyd.Device):\n",
+    "\n",
+    "    def kickoff(self):\n",
+    "        kickoff_status = ophyd.DeviceStatus(self)\n",
```

Comment:
it could look nicer to use:
```py
kickoff_status = ophyd.status.Status()
kickoff_status._finished(success=True)
```
since it's not doing anything.

Reply:
That's a great distinction that is of tutorial value.

Reply:
Yet, the distinction starts out as confusing.  Since *self* is a subclass of `Device`, why not use `DeviceStatus`?  This is the [search](http://nsls-ii.github.io/ophyd/search.html?q=status&check_keywords=yes&area=default) I start with to learn.  Nothing obvious near the top of the results list.  Needs insider knowledge to be a student now.  Seems `StatusBase` has two possibilities: `Status` and `DeviceStatus`.  The latter allows for a `MoveStatus`.  Lots to learn.

`tl;dr` <br />Start with the `Status` object.  Upgrade when needed.  This is a good learning point.

Reply:
The difference between `Status` and other variants (`DeviceStatus`, `MoveStatus`, etc.) is all about providing useful info for debugging and progress bars. If a status object "fails" -- i.e. the thread monitoring its progress calls `status._finshed(success=False)` -- a simple `Status` object can't even tell you which Device failed. `DeviceStatus` can. If a status object fails because the motor died, `MoveStatus` can tell you more detail about that. The essential interface is all the same; it's just about lumping in some metadata to help unwind failure modes.

Reply:
...so unless I'm doing a _really basic_ example focused on the Status API, I would general prefer `DeviceStatus` to `Status`.